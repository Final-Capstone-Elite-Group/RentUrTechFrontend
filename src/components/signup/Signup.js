import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { postUserToAPI } from '../../redux/user/user';
import 'react-toastify/dist/ReactToastify.css';
import style from '../../sass/shared/form.module.scss';

const Signup = () => {
  const dispatch = useDispatch();

  const [user, setUser] = useState({
    name: '',
    username: '',
    email: '',
    password: '',
  });

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(postUserToAPI(user));
    setUser({
      username: '',
      name: '',
      email: '',
      password: '',
    });
  };

  return (
    <>
      <div className={style.wrapper}>
        <form onSubmit={handleSubmit} className={style['user-form']}>
          <div>
            <h1>Sign up</h1>
            <div className={style.logo} />
          </div>
          <div className={style['form-group']}>
            <span htmlFor="name">
              Name
            </span>
            <input type="text" id="name" name="name" value={user.name} onChange={(e) => setUser({ ...user, name: e.target.value })} required />
          </div>
          <div className={style['form-group']}>
            <span htmlFor="username">
              Username
            </span>
            <input type="text" id="username" name="username" value={user.username} onChange={(e) => setUser({ ...user, username: e.target.value })} required />
          </div>
          <div className={style['form-group']}>
            <span htmlFor="email">
              Email
            </span>
            <input type="email" id="email" name="email" value={user.email} onChange={(e) => setUser({ ...user, email: e.target.value })} required />
          </div>
          <div className={style['form-group']}>
            <span htmlFor="password">
              Password
            </span>
            <input type="password" id="password" name="password" value={user.password} onChange={(e) => setUser({ ...user, password: e.target.value })} required />
          </div>
          <input type="submit" value="Sign up" className={style.submit} />
          <NavLink to="/login">login</NavLink>
        </form>
      </div>
    </>
  );
};

export default Signup;
