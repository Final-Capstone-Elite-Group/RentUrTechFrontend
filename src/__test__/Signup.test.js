import { render, screen, waitFor } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Signup from '../components/signup/Signup';
import store from '../redux/configure_store';

beforeEach(() => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <Signup />
      </BrowserRouter>
    </Provider>,
  );
});

test('Should match Login component snapshoot', async () => {
  const container = await waitFor(() => screen.getByText('login'));
  expect(container).toMatchSnapshot();
});

test('Should have a title Signup', () => {
  const signupTitle = screen.getByTestId('signup-title');
  expect(signupTitle).toBeInTheDocument();
});

test('Should have a email label', () => {
  const email = screen.getByTestId('email');
  expect(email).toBeInTheDocument();
});
