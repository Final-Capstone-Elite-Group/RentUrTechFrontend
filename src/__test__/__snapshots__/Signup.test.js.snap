// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match Login component snapshoot 1`] = `
<a
  class=""
  href="/login"
>
  login
</a>
`;
